examples/bq_file_load_benchmark/benchmark_tools/tables_processor.py

```diff
@@ -0,0 +1,161 @@
+            dataset_id,
+            bucket_name,
+            results_table_name,
+            results_table_dataset_id,
+            duplicate_benchmark_tables,
+            file_params,
+    ):
+        self.bq_project = bq_project
+        self.gcs_project = gcs_project
+        self.staging_project = staging_project
+        self.staging_dataset_id = staging_dataset_id
+        self.dataset_id = dataset_id
+        self.bucket_name = bucket_name
+        self.bucket_util = bucket_util.BucketUtil(
+            bucket_name=self.bucket_name,
+            project_id=self.gcs_project,
+            file_params=file_params,
+        )
+        self.results_table_name = results_table_name
+        self.results_table_dataset_id = results_table_dataset_id
+        self.duplicate_benchmark_tables = duplicate_benchmark_tables
+
+    def gather_files_with_benchmark_tables(self):
+        """Generates file combinations that already have benchmark tables.
+
+        Creates a set of files that already have been loaded to create
+        benchmark tables. Generates list by querying the job.sourceURI field
+        from the results table.
+        Returns:
+            Set of file names that already have been loaded to create
+            benchmark tables.
+        """
+        query = (
+            'SELECT job.sourceURI FROM `{0:s}.{1:s}.{2:s}` '.format(
+                self.bq_project,
+                self.results_table_dataset_id,
+                self.results_table_name,
+            )
+        )
+        query_job = bigquery.Client().query(
+            query,
+            location='US',
+        )
+        files_with_benchmark_tables = set()
+        for row in query_job:
+            if self.bucket_name in row['sourceURI']:
+                uri = row['sourceURI'].split('gs://{0:s}/'.format(
+                    self.bucket_name
+                ))[1]
+                file_name = uri.split('/*')[0]
+                files_with_benchmark_tables.add(file_name)
+        return files_with_benchmark_tables
+
+    def create_benchmark_tables(self):
+        """Creates a benchmark table for each file combination in GCS bucket.
+        """
+
+        # Gather files combinations that already have benchmark tables.
+        files_with_benchmark_tables = self.gather_files_with_benchmark_tables()
+        if self.duplicate_benchmark_tables:
+            files_to_skip = set()
+        else:
+            files_to_skip = files_with_benchmark_tables
+        # Gather file combinations that exist in the GCS Bucket.
+        existing_paths = self.bucket_util.get_existing_paths()
+        # Create a benchmark table for each existing file combination, and
+        # load the data from the file into the benchmark table.
+        for path in existing_paths:
+            path = path.split('/')
+            path = '/'.join(path[:len(path) - 1])
+            if path not in files_to_skip:
+
+                if path in files_with_benchmark_tables:
```

Comment:
nested if should be combined into single `if ... and ... :`

Reply:
I believe it's best as is here. The first if decides if a benchmark table needs to be created at all. The second if/else just decides if the benchmark table being created is a new table or a duplicate. Whether or not it's a new table or duplicate, it still needs to be created. So combining the if statements as suggested would look like: 

```
            if path not in files_to_skip and \
                    path in files_with_benchmark_tables:
                verb = 'Duplicating  '
            if path not in files_to_skip and \
                    path not in files_to_skip:
                verb = 'Processing '
            if path not in files_to_skip:
                logging.info('{0:s} benchmark table for {1:s}'.format(
                    verb,
                    path,
                ))
                table = benchmark_table.BenchmarkTable(
                    bq_project=self.bq_project,
                    gcs_project=self.gcs_project,
                    staging_project=self.staging_project,
                    staging_dataset_id=self.staging_dataset_id,
                    dataset_id=self.dataset_id,
                    bucket_name=self.bucket_name,
                    path=path,
                    results_table_name=self.results_table_name,
                    results_table_dataset_id=self.results_table_dataset_id,
                    bq_logs_dataset=self.bq_logs_dataset,
                )
                table.create_table()
                table.load_from_gcs()
```

Reply:
OR: 
     ```
 if path not in files_to_skip and \
                    path in files_with_benchmark_tables:
                verb = 'Duplicating  '
                logging.info('{0:s} benchmark table for {1:s}'.format(
                    verb,
                    path,
                ))
                table = benchmark_table.BenchmarkTable(
                    bq_project=self.bq_project,
                    gcs_project=self.gcs_project,
                    staging_project=self.staging_project,
                    staging_dataset_id=self.staging_dataset_id,
                    dataset_id=self.dataset_id,
                    bucket_name=self.bucket_name,
                    path=path,
                    results_table_name=self.results_table_name,
                    results_table_dataset_id=self.results_table_dataset_id,
                    bq_logs_dataset=self.bq_logs_dataset,
                )
                table.create_table()
                table.load_from_gcs()
            if path not in files_to_skip and \
                    path not in files_with_benchmark_tables:
                verb = 'Processing '
                logging.info('{0:s} benchmark table for {1:s}'.format(
                    verb,
                    path,
                ))
                table = benchmark_table.BenchmarkTable(
                    bq_project=self.bq_project,
                    gcs_project=self.gcs_project,
                    staging_project=self.staging_project,
                    staging_dataset_id=self.staging_dataset_id,
                    dataset_id=self.dataset_id,
                    bucket_name=self.bucket_name,
                    path=path,
                    results_table_name=self.results_table_name,
                    results_table_dataset_id=self.results_table_dataset_id,
                    bq_logs_dataset=self.bq_logs_dataset,
                )
                table.create_table()
                table.load_from_gcs()
```
But then that duplicates quite a bit of code

Reply:
ok you can leave this.

Reply:
ok leave this as is.